pkg/state/api.go

```diff
@@ -110,6 +109,9 @@ type StateInfo interface {
 	// True if state stores additional information in order to provide extended API.
 	ProvidesExtendedApi() (bool, error)
 
+	// Map on readable state. Way to apply multiple operations under same lock
+	MapR(func(StateInfo) (interface{}, error)) (interface{}, error)
```

Comment:
The naming is not good... Comment doesn't match with name. And `interface{}` everywhere, is it OK? Do we really need this function? Maybe a few of specific functions will be enough? How many different usages of this `MapR` do we have?